### PR TITLE
option to allow numeric slugs

### PR DESCRIPTION
**Pull request by d-m-**
Introduces an option `allow_numeric_slugs` in contenttypes.yml. 
See issues #3001 #4770 .

Docs (Contenttypes and records > Defining contenttypes > available options):

* `allow_numeric_slugs`: By default, Bolt prefixes slugs that would be purely numeric with the contenttype slug in order to prevent conflicts with routes of form /{contenttypeslug}/{id}. If this option is set to `true`, numeric slugs remain unprefixed. Care has to be taken not to use routes with IDs in the frontend.

Or something like this.

### CODE REVIEW

**Comment by bobdenotter** [comment]
I'd (personally) be hesitant to use this, but to each his own. If you use `/entries/3`, will the 'slug' or the 'id' get precedence? 

Anyhow, could you add a small explanation of this setting to the docs, at bolt/docs, please?

File: src/Legacy/Storage.php [hunk 1]
```diff
@@ -2795,8 +2795,8 @@ public function getUri($title, $id = 0, $contenttypeslug = "", $fulluri = true,
         $id = intval($id);
         $slug = $this->app['slugify']->slugify($title);
 
-        // Don't allow strictly numeric slugs.
-        if (is_numeric($slug)) {
+        // Don't allow strictly numeric slugs, unless allow_numeric_slugs options is set
+        if (is_numeric($slug) && $contenttype['allow_numeric_slugs'] != 'true') {
```
Comment (GawainLynch) [inline src/Legacy/Storage.php]: Can you please not use loose comparison, and `true` should not be a string but a boolean, e.g.
```php
if (is_numeric($slug) && $contenttype['allow_numeric_slugs'] !== true) {
```

**Comment by d-m-** [comment]
@GawainLynch You are totally right, i'll correct that, and add the explanation to the docs.
In particular, concerning your first comment, I'll add that the frontend router first tries slugs (which is the reason why this actually works and does not cause trouble unless a theme designer has chosen not to use record.link).

**Comment by d-m-** [comment]
@bobdenotter I am sorry that I forgot about the shorthand notation for setcontent queries, which always interprets numbers as ids. I added a check for the allow_numeric_slugs option to overwrite this behaviour. 

Basically I think that this is consistent: If one wants to use numeric slugs, it does not make sense to query 'contenttype/id', since the shorthands mimic the standard frontend routes.

**Comment by d-m-** [comment]
added docs: https://github.com/bolt/docs/pull/302

**Comment by bobdenotter** [comment]
@d-m-: Thank you for providing docs for this change! :+1:

**Comment by CarsonF** [comment]
Dang I wish I remembered to comment on this. This change was only applied to the *deprecated* code, hence the `Legacy` namespace. It needs to be written for the new storage code as well. And tests should be written as well. Else we are just providing ross with more work. Either way this pushes us farther away from release.

**Pull request merged**